finality-provider/store/fpstore_test.go

```diff
@@ -78,3 +79,128 @@ func FuzzFinalityProvidersStore(f *testing.F) {
 		require.ErrorIs(t, err, fpstore.ErrFinalityProviderNotFound)
 	})
 }
+
+func TestUpdateFpStatusFromVotingPower(t *testing.T) {
+	r := rand.New(rand.NewSource(10))
+
+	tcs := []struct {
+		name               string
+		fpStoredStatus     proto.FinalityProviderStatus
+		votingPowerOnChain uint64
+		expStatus          proto.FinalityProviderStatus
+		expErr             error
+	}{
+		{
+			"zero vp: Created to Registered",
+			proto.FinalityProviderStatus_CREATED,
+			0,
+			proto.FinalityProviderStatus_REGISTERED,
+			nil,
+		},
+		{
+			"zero vp: Active to Inactive",
+			proto.FinalityProviderStatus_ACTIVE,
+			0,
+			proto.FinalityProviderStatus_INACTIVE,
+			nil,
+		},
+		{
+			"zero vp: Slashed to Slashed",
+			proto.FinalityProviderStatus_SLASHED,
+			0,
+			proto.FinalityProviderStatus_SLASHED,
+			nil,
+		},
+		{
+			"vp > 0: Slashed to Active",
```

Comment:
agreed, modified to avoid updating the status if it is slashed at https://github.com/babylonlabs-io/finality-provider/pull/58/commits/d8fa3c6a46d0a6438c558f8a9e12ec6e73a69a08